File: library/src/com/google/maps/android/geojsonkmlabs/Point.java
package com.google.maps.android.geojsonkmlabs;

import com.google.android.gms.maps.model.LatLng;
/*
An abstraction that shares the common properties of KmlPoint and GeoJsonPoint
 */
public class Point implements Geometry {

    private final static String GEOMETRY_TYPE = "Point";

    private final LatLng mCoordinates;

    /**
     * Creates a new Point object
     *
     * @param coordinates coordinates of Point to store
     */
    public Point(LatLng coordinates) {
        if (coordinates == null) {
            throw new IllegalArgumentException("Coordinates cannot be null");
        }
        mCoordinates = coordinates;
    }

    /**
     * Gets the type of geometry
     *
     * @return type of geometry
     */
    public String getGeometryType() {
        return GEOMETRY_TYPE;
    }

    /**
     * Gets the coordinates of the Point
     *
     * @return coordinates of the Point
     */
    public LatLng getGeometryObject() {
        return mCoordinates;
    }

    /**
     * Gets the type of geometry. The type of geometry conforms to the GeoJSON 'type'
     * specification.
     *
     * @return type of geometry
     */
    public String getType() {
        return getGeometryType();
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder(GEOMETRY_TYPE).append("{");
        sb.append("\n coordinates=").append(mCoordinates);
        sb.append("\n}\n");
        return sb.toString();
    }

}

File: library/src/main/java/com/google/maps/android/data/Point.kt
/*
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.google.maps.android.data

import com.google.android.gms.maps.model.LatLng

/**
 * An abstraction that shares the common properties of
 * [com.google.maps.android.data.kml.KmlPoint] and
 * [com.google.maps.android.data.geojson.GeoJsonPoint]
 */
open class Point(coordinates: LatLng) : Geometry<LatLng> {

    private val _coordinates: LatLng = coordinates

    /**
     * Gets the coordinates of the Point
     */
    open val coordinates: LatLng
        get() = _coordinates

    /**
     * Gets the type of geometry
     */
    override val geometryType: String = "Point"

    /**
     * Gets the geometry object
     */
    override val geometryObject: LatLng = _coordinates

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Point

        return _coordinates == other._coordinates
    }

    override fun hashCode(): Int {
        return _coordinates.hashCode()
    }

    override fun toString(): String {
        return "Point(coordinates=$_coordinates)"
    }
}